src/render_helpers/mod.rs

```diff
@@ -255,27 +255,14 @@ pub fn render_to_shm(
     elements: impl Iterator<Item = impl RenderElement<GlesRenderer>>,
 ) -> anyhow::Result<()> {
     let _span = tracy_client::span!();
-
-    let buffer_size = buffer_dimensions(buffer).context("error getting buffer dimensions")?;
-    let size = buffer_size.to_logical(1, Transform::Normal).to_physical(1);
-
-    let mapping =
-        render_and_download(renderer, size, scale, transform, Fourcc::Argb8888, elements)?;
-    let bytes = renderer
-        .map_texture(&mapping)
-        .context("error mapping texture")?;
-
     shm::with_buffer_contents_mut(buffer, |shm_buffer, shm_len, buffer_data| {
-        ensure!(
-            // The buffer prefers pixels in little endian ...
-            buffer_data.format == wl_shm::Format::Argb8888
-                && buffer_data.stride == size.w * 4
-                && buffer_data.height == size.h
-                && shm_len as i32 == buffer_data.stride * buffer_data.height,
-            "invalid buffer format or size"
-        );
+        let size = Size::<_, Physical>::from((buffer_data.width, buffer_data.height));
+        let mapping =
+            render_and_download(renderer, size, scale, transform, Fourcc::Xrgb8888, elements)?;
 
-        ensure!(bytes.len() == shm_len, "mapped buffer has wrong length");
```

Comment:
These checks must remain here to ensure the unsafe code below doesn't access invalid memory

Reply:
This is currently only used by screencopy which does the checks already. Though indeed shm can be mutated anytime by the client. Again I am trying to avoid double checking the format and size, and I am still not convinced it is necessary to do so.

Reply:
This is a *safety* matter and we're writing Rust here :p Besides, these checks are trivial, there's no problem in checking them twice.

Reply:
Yes I agree. I will put them back in.